src/actions/node.actions.test.js

```diff
@@ -4,126 +4,141 @@ import thunk from 'redux-thunk';
+		const store = mockStore({
+			flowDesigner: {
+				nodes: new Map({ nodeId: { id: 'nodeId', type: 'type' } }),
+			},
 		});
 
-		it('setNodeGraphicalAttributes', () => {
-			const store = mockStore({
-				flowDesigner: {
-					nodes: new Map({ id: { id: 'nodeId', nodeType: 'type' } }),
-				},
-			});
+		store.dispatch(nodeActions.setNodeSize('nodeId', { width: 100, height: 100 }));
+		expect(store.getActions()).toMatchSnapshot();
+	});
 
-			store.dispatch(nodeActions.setNodeGraphicalAttributes('id', { selected: true }));
+	it('setNodeType', () => {
+		const nodeId = 'nodeId';
+		const nodeType = 'newNodeType';
+		const store = mockStore({
+			flowDesigner: {
+				nodes: new Map({ nodeId: { id: nodeId, nodeType: 'type' } }),
+			},
+		});
 
-			expect(store.getActions()).toMatchSnapshot();
+		store.dispatch(nodeActions.setNodeType(nodeId, nodeType));
+		expect(store.getActions()[0]).toEqual({
+			type: FLOWDESIGNER_NODE_SET_TYPE,
+			nodeId,
+			nodeType,
 		});
+	});
 
-		it('removeNodeGraphicalAttribute', () => {
-			const store = mockStore({
-				flowDesigner: {
-					nodes: new Map({ id: { id: 'nodeId', nodeType: 'type' } }),
-				},
-			});
+	it('setNodeGraphicalAttributes', () => {
+		const store = mockStore({
+			flowDesigner: {
+				nodes: new Map({ id: { id: 'nodeId', nodeType: 'type' } }),
+			},
+		});
+
+		store.dispatch(nodeActions.setNodeGraphicalAttributes('id', { selected: true }));
 
-			store.dispatch(nodeActions.removeNodeGraphicalAttribute('id', 'selected'));
+		expect(store.getActions()).toMatchSnapshot();
+	});
 
-			expect(store.getActions()).toMatchSnapshot();
+	it('removeNodeGraphicalAttribute', () => {
+		const store = mockStore({
+			flowDesigner: {
+				nodes: new Map({ id: { id: 'nodeId', nodeType: 'type' } }),
+			},
 		});
 
-		it('setNodeData', () => {
-			const store = mockStore({
-				flowDesigner: {
-					nodes: new Map({ id: { id: 'nodeId', nodeType: 'type' } }),
-				},
-			});
+		store.dispatch(nodeActions.removeNodeGraphicalAttribute('id', 'selected'));
 
-			store.dispatch(nodeActions.setNodeData('id', { type: 'test' }));
+		expect(store.getActions()).toMatchSnapshot();
+	});
 
-			expect(store.getActions()).toMatchSnapshot();
+	it('setNodeData', () => {
+		const store = mockStore({
+			flowDesigner: {
+				nodes: new Map({ id: { id: 'nodeId', nodeType: 'type' } }),
+			},
 		});
 
-		it('removeNodeData', () => {
-			const store = mockStore({
-				flowDesigner: {
-					nodes: new Map({
-						id: { id: 'nodeId', nodeType: 'type', data: new Map({ type: 'test' }) },
-					}),
-				},
-			});
+		store.dispatch(nodeActions.setNodeData('id', { type: 'test' }));
 
-			store.dispatch(nodeActions.removeNodeData('id', 'type'));
+		expect(store.getActions()).toMatchSnapshot();
+	});
 
-			expect(store.getActions()).toMatchSnapshot();
+	it('removeNodeData', () => {
+		const store = mockStore({
+			flowDesigner: {
+				nodes: new Map({
+					id: { id: 'nodeId', nodeType: 'type', data: new Map({ type: 'test' }) },
```

Comment:
same here.
data can have a type ?

Reply:
data can have any value it is a collection, changing name of the props to avoid confusion